diff --git a/Makefile b/Makefile
index 68b848a..79112c2 100644
--- a/Makefile
+++ b/Makefile
@@ -17,8 +17,8 @@ COMPRESS_XZ := y
 endif
 
 EXTRA_CFLAGS += -O2
-obj-m := rtlwifi.o
-rtlwifi-objs	:=	\
+obj-m := rtlwifi_new.o
+rtlwifi_new-objs	:=	\
 		base.o	\
 		cam.o	\
 		core.o	\
@@ -29,11 +29,11 @@ rtlwifi-objs	:=	\
 		regd.o	\
 		stats.o
 
-obj-m	+= rtl_pci.o
-rtl_pci-objs	:=		pci.o
+obj-m	+= rtl_pci_new.o
+rtl_pci_new-objs	:=		pci.o
 
-obj-m	+= rtl_usb.o
-rtl_usb-objs	:=		usb.o
+obj-m	+= rtl_usb_new.o
+rtl_usb_new-objs	:=		usb.o
 
 obj-m	+= btcoexist/
 obj-m	+= rtl8188ee/
@@ -68,20 +68,20 @@ endif
 	@mkdir -p $(MODDESTDIR)/rtl8723ae
 	@mkdir -p $(MODDESTDIR)/rtl8723be
 	@mkdir -p $(MODDESTDIR)/rtl8821ae
-	@install -p -D -m 644 rtl_pci.ko $(MODDESTDIR)	
-	@install -p -D -m 644 rtl_usb.ko $(MODDESTDIR)	
-	@install -p -D -m 644 rtlwifi.ko $(MODDESTDIR)
-	@install -p -D -m 644 ./btcoexist/btcoexist.ko $(MODDESTDIR)/btcoexist
-	@install -p -D -m 644 ./rtl8188ee/rtl8188ee.ko $(MODDESTDIR)/rtl8188ee
-	@install -p -D -m 644 ./rtl8192c/rtl8192c-common.ko $(MODDESTDIR)/rtl8192c
-	@install -p -D -m 644 ./rtl8192ce/rtl8192ce.ko $(MODDESTDIR)/rtl8192ce
-	@install -p -D -m 644 ./rtl8192cu/rtl8192cu.ko $(MODDESTDIR)/rtl8192cu
-	@install -p -D -m 644 ./rtl8192de/rtl8192de.ko $(MODDESTDIR)/rtl8192de
-	@install -p -D -m 644 ./rtl8192ee/rtl8192ee.ko $(MODDESTDIR)/rtl8192ee
-	@install -p -D -m 644 ./rtl8192se/rtl8192se.ko $(MODDESTDIR)/rtl8192se
-	@install -p -D -m 644 ./rtl8723ae/rtl8723ae.ko $(MODDESTDIR)/rtl8723ae
-	@install -p -D -m 644 ./rtl8723be/rtl8723be.ko $(MODDESTDIR)/rtl8723be
-	@install -p -D -m 644 ./rtl8821ae/rtl8821ae.ko $(MODDESTDIR)/rtl8821ae
+	@install -p -D -m 644 rtl_pci_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 rtl_usb_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 rtlwifi_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 ./btcoexist/btcoexist_new.ko $(MODDESTDIR)/btcoexist
+	@install -p -D -m 644 ./rtl8188ee/rtl8188ee_new.ko $(MODDESTDIR)/rtl8188ee
+	@install -p -D -m 644 ./rtl8192c/rtl8192c-common_new.ko $(MODDESTDIR)/rtl8192c
+	@install -p -D -m 644 ./rtl8192ce/rtl8192ce_new.ko $(MODDESTDIR)/rtl8192ce
+	@install -p -D -m 644 ./rtl8192cu/rtl8192cu_new.ko $(MODDESTDIR)/rtl8192cu
+	@install -p -D -m 644 ./rtl8192de/rtl8192de_new.ko $(MODDESTDIR)/rtl8192de
+	@install -p -D -m 644 ./rtl8192ee/rtl8192ee_new.ko $(MODDESTDIR)/rtl8192ee
+	@install -p -D -m 644 ./rtl8192se/rtl8192se_new.ko $(MODDESTDIR)/rtl8192se
+	@install -p -D -m 644 ./rtl8723ae/rtl8723ae_new.ko $(MODDESTDIR)/rtl8723ae
+	@install -p -D -m 644 ./rtl8723be/rtl8723be_new.ko $(MODDESTDIR)/rtl8723be
+	@install -p -D -m 644 ./rtl8821ae/rtl8821ae_new.ko $(MODDESTDIR)/rtl8821ae
 ifeq ($(COMPRESS_GZIP), y)
 	@gzip -f $(MODDESTDIR)/*.ko
 	@gzip -f $(MODDESTDIR)/btcoexist/*.ko
diff --git a/btcoexist/Makefile b/btcoexist/Makefile
index 17ecdad..b5d77e4 100644
--- a/btcoexist/Makefile
+++ b/btcoexist/Makefile
@@ -1,4 +1,4 @@
-btcoexist-objs :=	halbtc8723b2ant.o	\
+btcoexist_new-objs :=	halbtc8723b2ant.o	\
 			halbtcoutsrc.o		\
 			halbtc8192e2ant.o	\
 			halbtc8723b1ant.o	\
@@ -8,4 +8,4 @@ btcoexist-objs :=	halbtc8723b2ant.o	\
 			halbtcoutsrc.o		\
 			rtl_btc.o
 
-obj-m += btcoexist.o
+obj-m += btcoexist_new.o
diff --git a/rtl8188ee/Makefile b/rtl8188ee/Makefile
index 299bec3..ee5ddcb 100644
--- a/rtl8188ee/Makefile
+++ b/rtl8188ee/Makefile
@@ -1,17 +1,17 @@
-rtl8188ee-objs :=		\
-		dm.o		\
-		fw.o		\
-		hw.o		\
-		led.o		\
-		phy.o		\
-		pwrseq.o	\
-		pwrseqcmd.o	\
-		rf.o		\
-		sw.o		\
-		table.o		\
-		trx.o
-
-obj-m += rtl8188ee.o
-
-ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
-
+rtl8188ee_new-objs :=		\
+		dm.o		\
+		fw.o		\
+		hw.o		\
+		led.o		\
+		phy.o		\
+		pwrseq.o	\
+		pwrseqcmd.o	\
+		rf.o		\
+		sw.o		\
+		table.o		\
+		trx.o
+
+obj-m += rtl8188ee_new.o
+
+ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
+
diff --git a/rtl8192c/Makefile b/rtl8192c/Makefile
index ea0f8d2..47efd0b 100644
--- a/rtl8192c/Makefile
+++ b/rtl8192c/Makefile
@@ -1,9 +1,9 @@
-rtl8192c-common-objs :=		\
+rtl8192c-common_new-objs :=		\
 		main.o		\
 		dm_common.o	\
 		fw_common.o	\
 		phy_common.o
 
-obj-m += rtl8192c-common.o
+obj-m += rtl8192c-common_new.o
 
 ccflags-y += -D__CHECK_ENDIAN__
diff --git a/rtl8192ce/Makefile b/rtl8192ce/Makefile
index 8c3a1a3..0273769 100644
--- a/rtl8192ce/Makefile
+++ b/rtl8192ce/Makefile
@@ -1,4 +1,4 @@
-rtl8192ce-objs :=		\
+rtl8192ce_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hw.o		\
@@ -9,7 +9,7 @@ rtl8192ce-objs :=		\
 		table.o		\
 		trx.o
 
-obj-m += rtl8192ce.o
+obj-m += rtl8192ce_new.o
 
 ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
 
diff --git a/rtl8192cu/Makefile b/rtl8192cu/Makefile
index 2c16a98..01064ac 100644
--- a/rtl8192cu/Makefile
+++ b/rtl8192cu/Makefile
@@ -1,4 +1,4 @@
-rtl8192cu-objs :=		\
+rtl8192cu_new-objs :=		\
 		dm.o		\
 		hw.o		\
 		led.o		\
@@ -9,7 +9,7 @@ rtl8192cu-objs :=		\
 		table.o		\
 		trx.o
 
-obj-m += rtl8192cu.o
+obj-m += rtl8192cu_new.o
 
 ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
 
diff --git a/rtl8192de/Makefile b/rtl8192de/Makefile
index 23e7ed9..7a2bfbc 100644
--- a/rtl8192de/Makefile
+++ b/rtl8192de/Makefile
@@ -1,4 +1,4 @@
-rtl8192de-objs :=		\
+rtl8192de_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hw.o		\
@@ -9,7 +9,7 @@ rtl8192de-objs :=		\
 		table.o		\
 		trx.o
 
-obj-m += rtl8192de.o
+obj-m += rtl8192de_new.o
 
 ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
 
diff --git a/rtl8192ee/Makefile b/rtl8192ee/Makefile
index 909d1bc..53d658a 100644
--- a/rtl8192ee/Makefile
+++ b/rtl8192ee/Makefile
@@ -1,18 +1,18 @@
-rtl8192ee-objs :=		\
-		dm.o		\
-		fw.o		\
-		hw.o		\
-		led.o		\
-		phy.o		\
-		pwrseq.o	\
-		pwrseqcmd.o	\
-		rf.o		\
-		sw.o		\
-		table.o		\
-		trx.o
-
-obj-m += rtl8192ee.o
-
-ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
-
-
+rtl8192ee_new-objs :=		\
+		dm.o		\
+		fw.o		\
+		hw.o		\
+		led.o		\
+		phy.o		\
+		pwrseq.o	\
+		pwrseqcmd.o	\
+		rf.o		\
+		sw.o		\
+		table.o		\
+		trx.o
+
+obj-m += rtl8192ee_new.o
+
+ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
+
+
diff --git a/rtl8192se/Makefile b/rtl8192se/Makefile
index a0bd70c..cad2f0e 100644
--- a/rtl8192se/Makefile
+++ b/rtl8192se/Makefile
@@ -1,4 +1,4 @@
-rtl8192se-objs :=		\
+rtl8192se_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hw.o		\
@@ -9,7 +9,7 @@ rtl8192se-objs :=		\
 		table.o		\
 		trx.o
 
-obj-m += rtl8192se.o
+obj-m += rtl8192se_new.o
 
 ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
 
diff --git a/rtl8723ae/Makefile b/rtl8723ae/Makefile
index f322731..9ed3006 100644
--- a/rtl8723ae/Makefile
+++ b/rtl8723ae/Makefile
@@ -1,4 +1,4 @@
-rtl8723ae-objs :=		\
+rtl8723ae_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hal_btc.o	\
@@ -13,7 +13,7 @@ rtl8723ae-objs :=		\
 		table.o		\
 		trx.o		\
 
-obj-m += rtl8723ae.o
+obj-m += rtl8723ae_new.o
 
 ccflags-y += -Idrivers/net/wireless/rtlwifi -D__CHECK_ENDIAN__
 
diff --git a/rtl8723be/Makefile b/rtl8723be/Makefile
index 6719c1d..9581d60 100644
--- a/rtl8723be/Makefile
+++ b/rtl8723be/Makefile
@@ -1,4 +1,4 @@
-rtl8723be-objs :=		\
+rtl8723be_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hw.o		\
@@ -12,6 +12,6 @@ rtl8723be-objs :=		\
 		trx.o		\
 
 
-obj-m += rtl8723be.o
+obj-m += rtl8723be_new.o
 
 ccflags-y += -D__CHECK_ENDIAN__
diff --git a/rtl8821ae/Makefile b/rtl8821ae/Makefile
index 9b81b93..38de549 100644
--- a/rtl8821ae/Makefile
+++ b/rtl8821ae/Makefile
@@ -1,4 +1,4 @@
-rtl8821ae-objs :=		\
+rtl8821ae_new-objs :=		\
 		dm.o		\
 		fw.o		\
 		hw.o		\
@@ -12,6 +12,6 @@ rtl8821ae-objs :=		\
 		trx.o		\
 
 
-obj-m += rtl8821ae.o
+obj-m += rtl8821ae_new.o
 
 ccflags-y += -D__CHECK_ENDIAN__
diff --git a/tools/Makefile b/tools/Makefile
index 9eb0c49..64f5c67 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -18,7 +18,7 @@ endif
 
 EXTRA_CFLAGS += -O2
 EXTRA_CFLAGS += -DDEBUG
-obj-m := rtlwifi.o
+obj-m := rtlwifi_new.o
 PCI_MAIN_OBJS	:=	\
 		base.o	\
 		cam.o	\
@@ -30,13 +30,13 @@ PCI_MAIN_OBJS	:=	\
 		regd.o	\
 		stats.o
 
-obj-m	+= rtl_pci.o
-rtl_pci-objs	:=		pci.o
+obj-m	+= rtl_pci_new.o
+rtl_pci_new-objs	:=		pci.o
 
-obj-m	+= rtl_usb.o
-rtl_usb-objs	:=		usb.o
+obj-m	+= rtl_usb_new.o
+rtl_usb_new-objs	:=		usb.o
 
-rtlwifi-objs += $(PCI_MAIN_OBJS)
+rtlwifi_new-objs += $(PCI_MAIN_OBJS)
 
 ccflags-y += -D__CHECK_ENDIAN__
 
@@ -87,20 +87,20 @@ endif
 	@mkdir -p $(MODDESTDIR)/rtl8723ae
 	@mkdir -p $(MODDESTDIR)/rtl8723be
 	@mkdir -p $(MODDESTDIR)/rtl8821ae
-	@install -p -D -m 644 rtl_pci.ko $(MODDESTDIR)	
-	@install -p -D -m 644 rtl_usb.ko $(MODDESTDIR)	
-	@install -p -D -m 644 rtlwifi.ko $(MODDESTDIR)
-	@install -p -D -m 644 ./btcoexist/btcoexist.ko $(MODDESTDIR)/btcoexist
-	@install -p -D -m 644 ./rtl8188ee/rtl8188ee.ko $(MODDESTDIR)/rtl8188ee
-	@install -p -D -m 644 ./rtl8192c/rtl8192c-common.ko $(MODDESTDIR)/rtl8192c
-	@install -p -D -m 644 ./rtl8192ce/rtl8192ce.ko $(MODDESTDIR)/rtl8192ce
-	@install -p -D -m 644 ./rtl8192cu/rtl8192cu.ko $(MODDESTDIR)/rtl8192cu
-	@install -p -D -m 644 ./rtl8192de/rtl8192de.ko $(MODDESTDIR)/rtl8192de
-	@install -p -D -m 644 ./rtl8192ee/rtl8192ee.ko $(MODDESTDIR)/rtl8192ee
-	@install -p -D -m 644 ./rtl8192se/rtl8192se.ko $(MODDESTDIR)/rtl8192se
-	@install -p -D -m 644 ./rtl8723ae/rtl8723ae.ko $(MODDESTDIR)/rtl8723ae
-	@install -p -D -m 644 ./rtl8723be/rtl8723be.ko $(MODDESTDIR)/rtl8723be
-	@install -p -D -m 644 ./rtl8821ae/rtl8821ae.ko $(MODDESTDIR)/rtl8821ae
+	@install -p -D -m 644 rtl_pci_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 rtl_usb_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 rtlwifi_new.ko $(MODDESTDIR)
+	@install -p -D -m 644 ./btcoexist/btcoexist_new.ko $(MODDESTDIR)/btcoexist
+	@install -p -D -m 644 ./rtl8188ee/rtl8188ee_new.ko $(MODDESTDIR)/rtl8188ee
+	@install -p -D -m 644 ./rtl8192c/rtl8192c-common_new.ko $(MODDESTDIR)/rtl8192c
+	@install -p -D -m 644 ./rtl8192ce/rtl8192ce_new.ko $(MODDESTDIR)/rtl8192ce
+	@install -p -D -m 644 ./rtl8192cu/rtl8192cu_new.ko $(MODDESTDIR)/rtl8192cu
+	@install -p -D -m 644 ./rtl8192de/rtl8192de_new.ko $(MODDESTDIR)/rtl8192de
+	@install -p -D -m 644 ./rtl8192ee/rtl8192ee_new.ko $(MODDESTDIR)/rtl8192ee
+	@install -p -D -m 644 ./rtl8192se/rtl8192se_new.ko $(MODDESTDIR)/rtl8192se
+	@install -p -D -m 644 ./rtl8723ae/rtl8723ae_new.ko $(MODDESTDIR)/rtl8723ae
+	@install -p -D -m 644 ./rtl8723be/rtl8723be_new.ko $(MODDESTDIR)/rtl8723be
+	@install -p -D -m 644 ./rtl8821ae/rtl8821ae_new.ko $(MODDESTDIR)/rtl8821ae
 ifeq ($(COMPRESS_GZIP), y)
 	@gzip -f $(MODDESTDIR)/*.ko
 	@gzip -f $(MODDESTDIR)/btcoexist/*.ko
