diff --git a/rtl8188ee/sw.c b/rtl8188ee/sw.c
index c705c65..ea3ce88 100644
--- a/rtl8188ee/sw.c
+++ b/rtl8188ee/sw.c
@@ -170,7 +170,7 @@ int rtl88e_init_sw_vars(struct ieee80211_hw *hw)
 		return 1;
 	}
 
-	rtlpriv->cfg->fw_name = "rtlwifi/rtl8188efw.bin";
+	rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8188efw.bin";
 	rtlpriv->max_fw_size = 0x8000;
 	pr_info("Using firmware %s\n", rtlpriv->cfg->fw_name);
 	err = request_firmware_nowait(THIS_MODULE, 1, rtlpriv->cfg->fw_name,
@@ -287,7 +287,7 @@ static struct rtl_hal_cfg rtl88ee_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl88e_pci",
-	.fw_name = "rtlwifi/rtl8188efw.bin",
+	.fw_name = "rtlwifi_new/rtl8188efw.bin",
 	.ops = &rtl8188ee_hal_ops,
 	.mod_params = &rtl88ee_mod_params,
 
@@ -394,7 +394,7 @@ MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_AUTHOR("Larry Finger	<Larry.Finger@lwfinger.net>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8188E 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8188efw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8188efw.bin");
 
 module_param_named(swenc, rtl88ee_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl88ee_mod_params.debug, int, 0444);
diff --git a/rtl8192ce/sw.c b/rtl8192ce/sw.c
index 8bd9989..e143daa 100644
--- a/rtl8192ce/sw.c
+++ b/rtl8192ce/sw.c
@@ -162,9 +162,9 @@ int rtl92c_init_sw_vars(struct ieee80211_hw *hw)
 	/* request fw */
 	if (IS_VENDOR_UMC_A_CUT(rtlhal->version) &&
 	    !IS_92C_SERIAL(rtlhal->version))
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8192cfwU.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192cfwU.bin";
 	else if (IS_81xxC_VENDOR_UMC_B_CUT(rtlhal->version))
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8192cfwU_B.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192cfwU_B.bin";
 
 	rtlpriv->max_fw_size = 0x4000;
 	pr_info("Using firmware %s\n", rtlpriv->cfg->fw_name);
@@ -261,7 +261,7 @@ static struct rtl_hal_cfg rtl92ce_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl92c_pci",
-	.fw_name = "rtlwifi/rtl8192cfw.bin",
+	.fw_name = "rtlwifi_new/rtl8192cfw.bin",
 	.ops = &rtl8192ce_hal_ops,
 	.mod_params = &rtl92ce_mod_params,
 
@@ -367,9 +367,9 @@ MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_AUTHOR("Larry Finger	<Larry.Finger@lwfinger.net>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8192C/8188C 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8192cfw.bin");
-MODULE_FIRMWARE("rtlwifi/rtl8192cfwU.bin");
-MODULE_FIRMWARE("rtlwifi/rtl8192cfwU_B.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cfw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cfwU.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cfwU_B.bin");
 
 module_param_named(swenc, rtl92ce_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl92ce_mod_params.debug, int, 0444);
diff --git a/rtl8192cu/sw.c b/rtl8192cu/sw.c
index 57dd64e..9c72a6d 100644
--- a/rtl8192cu/sw.c
+++ b/rtl8192cu/sw.c
@@ -50,10 +50,10 @@ MODULE_AUTHOR("Ziv Huang	<ziv_huang@realtek.com>");
 MODULE_AUTHOR("Larry Finger	<Larry.Finger@lwfinger.net>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8192C/8188C 802.11n USB wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8192cufw.bin");
-MODULE_FIRMWARE("rtlwifi/rtl8192cufw_A.bin");
-MODULE_FIRMWARE("rtlwifi/rtl8192cufw_B.bin");
-MODULE_FIRMWARE("rtlwifi/rtl8192cufw_TMSC.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cufw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cufw_A.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cufw_B.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192cufw_TMSC.bin");
 
 static int rtl92cu_init_sw_vars(struct ieee80211_hw *hw)
 {
@@ -75,14 +75,14 @@ static int rtl92cu_init_sw_vars(struct ieee80211_hw *hw)
 	}
 	if (IS_VENDOR_UMC_A_CUT(rtlpriv->rtlhal.version) &&
 	    !IS_92C_SERIAL(rtlpriv->rtlhal.version)) {
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8192cufw_A.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192cufw_A.bin";
 	} else if (IS_81XXC_VENDOR_UMC_B_CUT(rtlpriv->rtlhal.version)) {
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8192cufw_B.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192cufw_B.bin";
 	} else {
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8192cufw_TMSC.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192cufw_TMSC.bin";
 	}
 	/* provide name of alternative file */
-	rtlpriv->cfg->alt_fw_name = "rtlwifi/rtl8192cufw.bin";
+	rtlpriv->cfg->alt_fw_name = "rtlwifi_new/rtl8192cufw.bin";
 	pr_info("Loading firmware %s\n", rtlpriv->cfg->fw_name);
 	rtlpriv->max_fw_size = 0x4000;
 	err = request_firmware_nowait(THIS_MODULE, 1,
@@ -184,7 +184,7 @@ static struct rtl_hal_usbint_cfg rtl92cu_interface_cfg = {
 
 static struct rtl_hal_cfg rtl92cu_hal_cfg = {
 	.name = "rtl92c_usb",
-	.fw_name = "rtlwifi/rtl8192cufw.bin",
+	.fw_name = "rtlwifi_new/rtl8192cufw.bin",
 	.ops = &rtl8192cu_hal_ops,
 	.mod_params = &rtl92cu_mod_params,
 	.usb_interface_cfg = &rtl92cu_interface_cfg,
diff --git a/rtl8192de/sw.c b/rtl8192de/sw.c
index ebd7ff9..54d77c4 100644
--- a/rtl8192de/sw.c
+++ b/rtl8192de/sw.c
@@ -277,7 +277,7 @@ static struct rtl_hal_cfg rtl92de_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl92d_pci",
-	.fw_name = "rtlwifi/rtl8192defw.bin",
+	.fw_name = "rtlwifi_new/rtl8192defw.bin",
 	.ops = &rtl8192de_hal_ops,
 	.mod_params = &rtl92de_mod_params,
 
@@ -380,7 +380,7 @@ MODULE_AUTHOR("lizhaoming	<chaoming_li@realsil.com.cn>");
 MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8192DE 802.11n Dul Mac PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8192defw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192defw.bin");
 
 module_param_named(swenc, rtl92de_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl92de_mod_params.debug, int, 0444);
diff --git a/rtl8192ee/sw.c b/rtl8192ee/sw.c
index 068d31e..2fe68e6 100644
--- a/rtl8192ee/sw.c
+++ b/rtl8192ee/sw.c
@@ -180,7 +180,7 @@ int rtl92ee_init_sw_vars(struct ieee80211_hw *hw)
 	}
 
 	/* request fw */
-	rtlpriv->cfg->fw_name = "rtlwifi/rtl8192eefw_new.bin";
+	rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8192eefw_new.bin";
 
 	rtlpriv->max_fw_size = 0x8000;
 	pr_info("Using firmware %s\n", rtlpriv->cfg->fw_name);
@@ -277,7 +277,7 @@ static struct rtl_hal_cfg rtl92ee_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl92ee_pci",
-	.fw_name = "rtlwifi/rtl8192eefw.bin",
+	.fw_name = "rtlwifi_new/rtl8192eefw.bin",
 	.ops = &rtl8192ee_hal_ops,
 	.mod_params = &rtl92ee_mod_params,
 
@@ -381,7 +381,7 @@ MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_AUTHOR("Larry Finger	<Larry.Finger@lwfinger.net>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8192EE 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8192eefw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192eefw.bin");
 
 module_param_named(swenc, rtl92ee_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl92ee_mod_params.debug, int, 0444);
diff --git a/rtl8192se/hw.c b/rtl8192se/hw.c
index d6761b1..040dc25 100644
--- a/rtl8192se/hw.c
+++ b/rtl8192se/hw.c
@@ -1009,7 +1009,7 @@ int rtl92se_hw_init(struct ieee80211_hw *hw)
 	if (!rtstatus) {
 		RT_TRACE(rtlpriv, COMP_ERR, DBG_WARNING, "Failed to download FW. "
 				"Init HW without FW now.., Please copy FW into"
-				"/lib/firmware/rtlwifi\n");
+				"/lib/firmware/rtlwifi_new\n");
 		rtlhal->fw_ready = false;
 	} else {
 		rtlhal->fw_ready = true;
diff --git a/rtl8192se/sw.c b/rtl8192se/sw.c
index e99fb3e..b7b677e 100644
--- a/rtl8192se/sw.c
+++ b/rtl8192se/sw.c
@@ -303,7 +303,7 @@ static struct rtl_hal_cfg rtl92se_hal_cfg = {
 	.bar_id = 1,
 	.write_readback = false,
 	.name = "rtl92s_pci",
-	.fw_name = "rtlwifi/rtl8192sefw.bin",
+	.fw_name = "rtlwifi_new/rtl8192sefw.bin",
 	.ops = &rtl8192se_hal_ops,
 	.mod_params = &rtl92se_mod_params,
 
@@ -410,7 +410,7 @@ MODULE_AUTHOR("lizhaoming	<chaoming_li@realsil.com.cn>");
 MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8192S/8191S 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8192sefw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8192sefw.bin");
 
 module_param_named(swenc, rtl92se_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl92se_mod_params.debug, int, 0444);
diff --git a/rtl8723ae/sw.c b/rtl8723ae/sw.c
index 06fd29a..130475d 100644
--- a/rtl8723ae/sw.c
+++ b/rtl8723ae/sw.c
@@ -175,9 +175,9 @@ int rtl8723e_init_sw_vars(struct ieee80211_hw *hw)
 	}
 
 	if (IS_VENDOR_8723_A_CUT(rtlhal->version))
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8723fw.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8723fw.bin";
 	else if (IS_81xxC_VENDOR_UMC_B_CUT(rtlhal->version))
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8723fw_B.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8723fw_B.bin";
 
 	rtlpriv->max_fw_size = 0x6000;
 	pr_info("Using firmware %s\n", rtlpriv->cfg->fw_name);
@@ -271,7 +271,7 @@ static struct rtl_hal_cfg rtl8723e_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl8723e_pci",
-	.fw_name = "rtlwifi/rtl8723efw.bin",
+	.fw_name = "rtlwifi_new/rtl8723efw.bin",
 	.ops = &rtl8723e_hal_ops,
 	.mod_params = &rtl8723e_mod_params,
 	.maps[SYS_ISO_CTRL] = REG_SYS_ISO_CTRL,
@@ -374,7 +374,7 @@ MODULE_AUTHOR("lizhaoming	<chaoming_li@realsil.com.cn>");
 MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8723E 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8723efw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8723efw.bin");
 
 module_param_named(swenc, rtl8723e_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl8723e_mod_params.debug, int, 0444);
diff --git a/rtl8723be/sw.c b/rtl8723be/sw.c
index f02b1d8..af91396 100644
--- a/rtl8723be/sw.c
+++ b/rtl8723be/sw.c
@@ -272,7 +272,7 @@ static struct rtl_hal_cfg rtl8723be_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl8723be_pci",
-	.fw_name = "rtlwifi/rtl8723befw.bin",
+	.fw_name = "rtlwifi_new/rtl8723befw.bin",
 	.ops = &rtl8723be_hal_ops,
 	.mod_params = &rtl8723be_mod_params,
 	.maps[SYS_ISO_CTRL] = REG_SYS_ISO_CTRL,
@@ -377,7 +377,7 @@ MODULE_AUTHOR("PageHe	<page_he@realsil.com.cn>");
 MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8723BE 802.11n PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8723befw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8723befw.bin");
 
 module_param_named(swenc, rtl8723be_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl8723be_mod_params.debug, int, 0444);
diff --git a/rtl8821ae/sw.c b/rtl8821ae/sw.c
index 21ba8ed..f075e75 100644
--- a/rtl8821ae/sw.c
+++ b/rtl8821ae/sw.c
@@ -106,9 +106,9 @@ static void load_wowlan_fw(struct rtl_priv *rtlpriv)
 	}
 
 	if (rtlpriv->rtlhal.hw_type == HARDWARE_TYPE_RTL8821AE)
-		fw_name = "rtlwifi/rtl8821aefw_wowlan.bin";
+		fw_name = "rtlwifi_new/rtl8821aefw_wowlan.bin";
 	else
-		fw_name = "rtlwifi/rtl8812aefw_wowlan.bin";
+		fw_name = "rtlwifi_new/rtl8812aefw_wowlan.bin";
 	err = request_firmware(&wowlan_firmware, fw_name, rtlpriv->io.dev);
 	if (err) {
 		RT_TRACE(rtlpriv, COMP_ERR, DBG_EMERG,
@@ -244,9 +244,9 @@ int rtl8821ae_init_sw_vars(struct ieee80211_hw *hw)
 	}
 
 	if (rtlhal->hw_type == HARDWARE_TYPE_RTL8812AE)
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8812aefw.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8812aefw.bin";
 	else
-		rtlpriv->cfg->fw_name = "rtlwifi/rtl8821aefw.bin";
+		rtlpriv->cfg->fw_name = "rtlwifi_new/rtl8821aefw.bin";
 
 	rtlpriv->max_fw_size = 0x8000;
 	pr_info("Using firmware %s\n", rtlpriv->cfg->fw_name);
@@ -344,7 +344,7 @@ static struct rtl_hal_cfg rtl8821ae_hal_cfg = {
 	.bar_id = 2,
 	.write_readback = true,
 	.name = "rtl8821ae_pci",
-	.fw_name = "rtlwifi/rtl8821aefw.bin",
+	.fw_name = "rtlwifi_new/rtl8821aefw.bin",
 	.ops = &rtl8821ae_hal_ops,
 	.mod_params = &rtl8821ae_mod_params,
 	.maps[SYS_ISO_CTRL] = REG_SYS_ISO_CTRL,
@@ -456,7 +456,7 @@ MODULE_DEVICE_TABLE(pci, rtl8821ae_pci_ids);
 MODULE_AUTHOR("Realtek WlanFAE	<wlanfae@realtek.com>");
 MODULE_LICENSE("GPL");
 MODULE_DESCRIPTION("Realtek 8821ae 802.11ac PCI wireless");
-MODULE_FIRMWARE("rtlwifi/rtl8821aefw.bin");
+MODULE_FIRMWARE("rtlwifi_new/rtl8821aefw.bin");
 
 module_param_named(swenc, rtl8821ae_mod_params.sw_crypto, bool, 0444);
 module_param_named(debug, rtl8821ae_mod_params.debug, int, 0444);
