--- CMake/Packages/FindOGRE.cmake.orig	2011-01-04 02:41:08.000000000 -0600
+++ CMake/Packages/FindOGRE.cmake	2011-01-04 02:42:16.000000000 -0600
@@ -259,6 +259,12 @@
 	endif ()
   endif ()
   
+  if (NOT OGRE_DEPS_FOUND)
+    pkg_message(OGRE "Could not find all required dependencies for the Ogre package.")
+    set(OGRE_FOUND FALSE)
+  endif ()
+endif ()
+
   if (OGRE_CONFIG_THREADS)
     if (OGRE_CONFIG_THREAD_PROVIDER EQUAL 1)
       find_package(Boost COMPONENTS thread QUIET)
@@ -284,12 +290,6 @@
     endif ()
   endif ()
   
-  if (NOT OGRE_DEPS_FOUND)
-    pkg_message(OGRE "Could not find all required dependencies for the Ogre package.")
-    set(OGRE_FOUND FALSE)
-  endif ()
-endif ()
-
 if (NOT OGRE_FOUND)
   return()
 endif ()
@@ -485,4 +485,4 @@
 find_path(OGRE_MEDIA_DIR NAMES packs/OgreCore.zip HINTS ${OGRE_MEDIA_SEARCH_PATH}
   PATHS ${OGRE_PREFIX_PATH} PATH_SUFFIXES ${OGRE_MEDIA_SEARCH_SUFFIX})
 
-endif () # if from line 51-56
\ No newline at end of file
+endif () # if from line 51-56
