diff -up ogre-1.7.2/CMake/Templates/OGRE-PCZ.pc.in.rpath ogre-1.7.2/CMake/Templates/OGRE-PCZ.pc.in
--- ogre-1.7.2/CMake/Templates/OGRE-PCZ.pc.in.rpath	2010-12-20 14:34:20.659800000 -0500
+++ ogre-1.7.2/CMake/Templates/OGRE-PCZ.pc.in	2010-12-20 14:34:25.549800001 -0500
@@ -7,5 +7,5 @@ Name: OGRE-PCZ
 Description: Portal-Connected Zone Scene Manager for OGRE
 Version: @OGRE_VERSION@
 Requires: OGRE = @OGRE_VERSION@
-Libs: ${libdir}/OGRE/@OGRE_PLUGIN_PREFIX@Plugin_PCZSceneManager@OGRE_LIB_SUFFIX@@OGRE_PLUGIN_EXT@ ${libdir}/OGRE/@OGRE_PLUGIN_PREFIX@Plugin_OctreeZone@OGRE_LIB_SUFFIX@@OGRE_PLUGIN_EXT@ -Wl,-rpath,${libdir}/OGRE
+Libs: ${libdir}/OGRE/@OGRE_PLUGIN_PREFIX@Plugin_PCZSceneManager@OGRE_LIB_SUFFIX@@OGRE_PLUGIN_EXT@ ${libdir}/OGRE/@OGRE_PLUGIN_PREFIX@Plugin_OctreeZone@OGRE_LIB_SUFFIX@@OGRE_PLUGIN_EXT@
 Cflags: -I${includedir}/OGRE/Plugins/PCZSceneManager -I${includedir}/OGRE/Plugins/OctreeZone @OGRE_CFLAGS@
