diff --git a/pgmodeler.pri b/pgmodeler.pri
index 06edd7b..49a7ae9 100644
--- a/pgmodeler.pri
+++ b/pgmodeler.pri
@@ -229,12 +229,12 @@ defineTest(printBuildDetails) {
  log("  PRIVATELIBDIR = $$PRIVATELIBDIR $$LB")
  log("  PLUGINSDIR    = $$PLUGINSDIR $$LB")
  log("  SHAREDIR      = $$SHAREDIR $$LB")
- log("  CONFDIR       = $$SHAREDIR/conf $$LB")
- log("  DOCDIR        = $$SHAREDIR/doc $$LB")
- log("  LANGDIR       = $$SHAREDIR/lang $$LB")
- log("  SAMPLESDIR    = $$SHAREDIR/samples $$LB")
- log("  SCHEMASDIR    = $$SHAREDIR/schemas $$LB")
- log("  TEMPDIR       = $$PREFIX/tmp $$LB $$LB")
+ log("  CONFDIR       = $$CONFDIR $$LB")
+ log("  DOCDIR        = $$DOCDIR $$LB")
+ log("  LANGDIR       = $$LANGDIR $$LB")
+ log("  SAMPLESDIR    = $$SAMPLESDIR $$LB")
+ log("  SCHEMASDIR    = $$SCHEMASDIR $$LB")
+ log("  TEMPDIR       = $$TEMPDIR $$LB $$LB")
  log("* To change a variable value run qmake again setting the desired value e.g.: $$LB")
  log("  > qmake PREFIX+=/usr/local -r pgmodeler.pro $$LB $$LB")
  log("* Proceed with build process by running: $$LB")
