diff -NaurEbBH -x '*.so' -x '*.so.*' -x '*.o' pgmodeler.orig/libpgmodeler/src/baseobject.cpp pgmodeler/libpgmodeler/src/baseobject.cpp
--- pgmodeler.orig/libpgmodeler/src/baseobject.cpp	2013-06-12 22:30:51.000000000 +0400
+++ pgmodeler/libpgmodeler/src/baseobject.cpp	2013-06-12 23:25:21.311156833 +0400
@@ -1,4 +1,4 @@
-/*
+/*
 # PostgreSQL Database Modeler (pgModeler)
 #
 # Copyright 2006-2013 - Raphael Araújo e Silva <rkhaotix@gmail.com>
@@ -81,7 +81,7 @@
 	attributes[ParsersAttributes::COLLATION]="";
 	attributes[ParsersAttributes::PROTECTED]="";
 	attributes[ParsersAttributes::SQL_DISABLED]="";
-	this->setName(QApplication::translate("BaseObject","new_object","", -1));
+	this->setName(QApplication::translate("BaseObject","new_object","", QCoreApplication::UnicodeUTF8));
 }
 
 unsigned BaseObject::getGlobalId(void)
@@ -95,7 +95,7 @@
 		/* Due to the class BaseObject not be derived from QObject the function tr() is inefficient to
 		 translate the type names thus the method called to do the translation is from the application
 		 specifying the context (BaseObject) in the ts file and the text to be translated */
-		return(QApplication::translate("BaseObject",obj_type_names[obj_type].toStdString().c_str(),"", -1));
+		return(QApplication::translate("BaseObject",obj_type_names[obj_type].toStdString().c_str(),"", QCoreApplication::UnicodeUTF8));
 	else
 		return("");
 }
diff -NaurEbBH -x '*.so' -x '*.so.*' -x '*.o' pgmodeler.orig/libutils/src/exception.cpp pgmodeler/libutils/src/exception.cpp
--- pgmodeler.orig/libutils/src/exception.cpp	2013-06-12 22:30:51.000000000 +0400
+++ pgmodeler/libutils/src/exception.cpp	2013-06-12 23:09:30.086415982 +0400
@@ -235,7 +235,7 @@
 	/* Because the Exception class is not derived from QObject the function tr() is inefficient to translate messages
 		so the translation method is called  directly from the application specifying the
 		context (Exception) in the ts file and the text to be translated */
-	configureException(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"", -1),
+	configureException(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"", QCoreApplication::UnicodeUTF8),
 										 error_type, method, file, line, extra_info);
 
 	if(exception) addException(*exception);
@@ -254,7 +254,7 @@
 	/* Because the Exception class is not derived from QObject the function tr() is inefficient to translate messages
 		so the translation method is called  directly from the application specifying the
 		context (Exception) in the ts file and the text to be translated */
-	configureException(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"",-1),
+	configureException(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"",QCoreApplication::UnicodeUTF8),
 										 error_type, method, file, line, extra_info);
 
 	itr=exceptions.begin();
@@ -302,7 +302,7 @@
 		/* Because the Exception class is not derived from QObject the function tr() is inefficient to translate messages
 		 so the translation method is called  directly from the application specifying the
 		 context (Exception) in the ts file and the text to be translated */
-		return(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"", -1));
+		return(QApplication::translate("Exception",messages[error_type][ERROR_MESSAGE].toStdString().c_str(),"", QCoreApplication::UnicodeUTF8));
 	else
 		return("");
 }
