diff -NurEbBH rigs-of-rods-23c619a9010e375ea7357ef190ff683f2beac5d4.orig/source/main/RoRPrerequisites.h rigs-of-rods-23c619a9010e375ea7357ef190ff683f2beac5d4/source/main/RoRPrerequisites.h
--- rigs-of-rods-23c619a9010e375ea7357ef190ff683f2beac5d4.orig/source/main/RoRPrerequisites.h	2016-01-07 23:27:32.000000000 +0300
+++ rigs-of-rods-23c619a9010e375ea7357ef190ff683f2beac5d4/source/main/RoRPrerequisites.h	2016-01-17 20:21:46.465101398 +0300
@@ -60,7 +60,7 @@
 
 #include <MyGUI_Prerequest.h> // Forward declarations
 
-#if MYGUI_VERSION >= 0x030202
+#if MYGUI_VERSION >= 0x030201
 #	define MYGUI_GET_SCANCODE(KEY) (KEY.getValue())
 #else
 #	define MYGUI_GET_SCANCODE(KEY) (KEY.toValue())
