diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gameplay/ChatSystem.cpp rigsofrods-source-0.4.0.4/source/main/gameplay/ChatSystem.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/gameplay/ChatSystem.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gameplay/ChatSystem.cpp	2012-09-23 18:11:04.754067350 +0400
@@ -25,6 +25,7 @@
 #include "Network.h"
 #include "PlayerColours.h"
 #include "Utils.h"
+#include "SkyManager.h"
 
 #ifdef USE_MYGUI
 #include "GUIMp.h"
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gameplay/Landusemap.cpp rigsofrods-source-0.4.0.4/source/main/gameplay/Landusemap.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/gameplay/Landusemap.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gameplay/Landusemap.cpp	2012-09-23 18:11:04.772067642 +0400
@@ -28,8 +28,8 @@
 #include "TerrainManager.h"
 
 #ifdef USE_PAGED
-#include "PropertyMaps.h"
-#include "PagedGeometry.h"
+#include "OGRE/PagedGeometry/PropertyMaps.h"
+#include "OGRE/PagedGeometry/PagedGeometry.h"
 #endif
 
 using namespace Ogre;
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gameplay/OutProtocol.h rigsofrods-source-0.4.0.4/source/main/gameplay/OutProtocol.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gameplay/OutProtocol.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gameplay/OutProtocol.h	2012-09-23 18:11:04.879069384 +0400
@@ -27,7 +27,7 @@
 #include "Singleton.h"
 
 #ifdef USE_SOCKETW
-#include "SocketW.h"
+#include "SocketW/SocketW.h"
 #endif //USE_SOCKETW
 
 #ifdef WIN32
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gameplay/Replay.h rigsofrods-source-0.4.0.4/source/main/gameplay/Replay.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gameplay/Replay.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gameplay/Replay.h	2012-09-23 18:11:04.751067302 +0400
@@ -25,7 +25,7 @@
 #include "Beam.h"
 
 #ifdef USE_MYGUI
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 #endif //MYGUI
 
 
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gameplay/SceneMouse.cpp rigsofrods-source-0.4.0.4/source/main/gameplay/SceneMouse.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/gameplay/SceneMouse.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gameplay/SceneMouse.cpp	2012-09-24 17:20:02.145232882 +0400
@@ -24,7 +24,7 @@
 #include "CameraManager.h"
 
 #ifdef USE_MYGUI
-# include <MyGUI.h>
+# include <MYGUI/MyGUI.h>
 #endif //USE_MYGUI
 
 using namespace Ogre;
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/DashBoardManager.h rigsofrods-source-0.4.0.4/source/main/gui/DashBoardManager.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/DashBoardManager.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/DashBoardManager.h	2012-09-23 18:11:04.559064177 +0400
@@ -25,7 +25,7 @@
 #include "RoRPrerequisites.h"
 #include "Singleton.h"
 
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 
 // TODO: Clean up this header
 
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIFriction.h rigsofrods-source-0.4.0.4/source/main/gui/GUIFriction.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIFriction.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/GUIFriction.h	2012-09-23 18:11:04.630065331 +0400
@@ -25,7 +25,7 @@
 #include "RoRPrerequisites.h"
 #include "Singleton.h"
 
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 
 class GUI_Friction : public RoRSingleton < GUI_Friction >, public ZeroedMemoryAllocator
 {
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIManager.cpp rigsofrods-source-0.4.0.4/source/main/gui/GUIManager.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIManager.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/GUIManager.cpp	2012-09-23 18:11:04.655065740 +0400
@@ -30,7 +30,7 @@
 #include "Settings.h"
 #include "TerrainManager.h"
 
-#include <MyGUI_OgrePlatform.h>
+#include <MYGUI/MyGUI_OgrePlatform.h>
 
 using namespace Ogre;
 
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIManager.h rigsofrods-source-0.4.0.4/source/main/gui/GUIManager.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIManager.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/GUIManager.h	2012-09-23 18:11:04.592064714 +0400
@@ -27,7 +27,7 @@
 #include "Ogre.h"
 #include "Singleton.h"
 
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 
 #define GETMYGUI GUIManager::getSingleton().getGUI()
 
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIMenu.h rigsofrods-source-0.4.0.4/source/main/gui/GUIMenu.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIMenu.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/GUIMenu.h	2012-09-23 18:11:04.661065836 +0400
@@ -27,7 +27,7 @@
 #include "BeamData.h"
 #include "Singleton.h"
 
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 #include <pthread.h>
 
 class GUI_MainMenu : public RoRSingletonNoCreation< GUI_MainMenu >, public ZeroedMemoryAllocator
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIMp.h rigsofrods-source-0.4.0.4/source/main/gui/GUIMp.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/GUIMp.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/GUIMp.h	2012-09-23 18:11:04.689066292 +0400
@@ -28,7 +28,7 @@
 #include "rornet.h"
 #include "Singleton.h"
 
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 
 class GUI_Multiplayer : public RoRSingletonNoCreation< GUI_Multiplayer >
 {
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/mygui/BaseLayout.h rigsofrods-source-0.4.0.4/source/main/gui/mygui/BaseLayout.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/mygui/BaseLayout.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/mygui/BaseLayout.h	2012-09-23 18:11:04.589064666 +0400
@@ -28,7 +28,7 @@
 #define __BASE_LAYOUT_H__
 
 #include "RoRPrerequisites.h"
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 #include "WrapsAttribute.h"
 
 namespace wraps
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/gui/mygui/WrapsAttribute.h rigsofrods-source-0.4.0.4/source/main/gui/mygui/WrapsAttribute.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/mygui/WrapsAttribute.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/mygui/WrapsAttribute.h	2012-09-23 18:11:04.580064519 +0400
@@ -28,7 +28,7 @@
 #define __WRAPS_ATTRIBUTE_H__
 
 #include "RoRPrerequisites.h"
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 #include "Attribute.h"
 
 namespace attribute
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/network/Network.h rigsofrods-source-0.4.0.4/source/main/network/Network.h
--- rigsofrods-source-0.4.0.4.orig/source/main/network/Network.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/network/Network.h	2012-09-23 18:11:03.041039477 +0400
@@ -25,7 +25,7 @@
 
 #include "BeamData.h"
 #include "rornet.h"
-#include "SocketW.h"
+#include "SocketW/SocketW.h"
 
 #include <pthread.h>
 
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/network/NetworkStreamManager.h rigsofrods-source-0.4.0.4/source/main/network/NetworkStreamManager.h
--- rigsofrods-source-0.4.0.4.orig/source/main/network/NetworkStreamManager.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/network/NetworkStreamManager.h	2012-09-23 18:11:03.040039461 +0400
@@ -31,7 +31,7 @@
 #include "Singleton.h"
 
 #ifdef USE_SOCKETW
-#include "SocketW.h"
+#include "SocketW/SocketW.h"
 #endif // USE_SOCKETW
 
 class Streamable;
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/network/StreamableFactory.h rigsofrods-source-0.4.0.4/source/main/network/StreamableFactory.h
--- rigsofrods-source-0.4.0.4.orig/source/main/network/StreamableFactory.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/network/StreamableFactory.h	2012-09-23 18:11:03.050039625 +0400
@@ -31,7 +31,7 @@
 #include <pthread.h>
 
 #ifdef USE_SOCKETW
-#include "SocketW.h"
+#include "SocketW/SocketW.h"
 #endif //SOCKETW
 
 template<class T, class X> class StreamableFactory : public StreamableFactoryInterface
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/physics/mplatform/MPlatformFD.h rigsofrods-source-0.4.0.4/source/main/physics/mplatform/MPlatformFD.h
--- rigsofrods-source-0.4.0.4.orig/source/main/physics/mplatform/MPlatformFD.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/physics/mplatform/MPlatformFD.h	2012-09-23 18:11:02.140024817 +0400
@@ -23,7 +23,7 @@
 
 #include "RoRPrerequisites.h"
 #include "MPlatformBase.h"
-#include "SocketW.h"
+#include "SocketW/SocketW.h"
 
 class MPlatform_FD : public MPlatform_Base
 {
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/terrain/TerrainManager.cpp rigsofrods-source-0.4.0.4/source/main/terrain/TerrainManager.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/terrain/TerrainManager.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/terrain/TerrainManager.cpp	2012-09-24 17:26:51.174881583 +0400
@@ -30,7 +30,7 @@
 #include "HDRListener.h"
 #include "Language.h"
 #include "RoRFrameListener.h"
-#include "ScriptEngine.h"
+#include "scripting/ScriptEngine.h"
 #include "Settings.h"
 #include "ShadowManager.h"
 #include "SkyManager.h"
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/terrain/TerrainObjectManager.h rigsofrods-source-0.4.0.4/source/main/terrain/TerrainObjectManager.h
--- rigsofrods-source-0.4.0.4.orig/source/main/terrain/TerrainObjectManager.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/terrain/TerrainObjectManager.h	2012-09-23 18:11:02.307027534 +0400
@@ -25,12 +25,12 @@
 #include "IManager.h"
 
 #ifdef USE_PAGED
-#include "BatchPage.h"
-#include "GrassLoader.h"
-#include "ImpostorPage.h"
-#include "PagedGeometry.h"
-#include "TreeLoader2D.h"
-#include "TreeLoader3D.h"
+#include "OGRE/PagedGeometry/BatchPage.h"
+#include "OGRE/PagedGeometry/GrassLoader.h"
+#include "OGRE/PagedGeometry/ImpostorPage.h"
+#include "OGRE/PagedGeometry/PagedGeometry.h"
+#include "OGRE/PagedGeometry/TreeLoader2D.h"
+#include "OGRE/PagedGeometry/TreeLoader3D.h"
 #endif //USE_PAGED
 
 class TerrainObjectManager : public IManager
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/utils/Language.cpp rigsofrods-source-0.4.0.4/source/main/utils/Language.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/utils/Language.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/utils/Language.cpp	2012-09-23 18:11:02.232026315 +0400
@@ -26,10 +26,10 @@
 #include "Settings.h"
 
 #ifdef USE_MYGUI
-#include <MyGUI.h>
-#include <MyGUI_IFont.h>
-#include <MyGUI_FontData.h>
-#include <MyGUI_FontManager.h>
+#include <MYGUI/MyGUI.h>
+#include <MYGUI/MyGUI_IFont.h>
+#include <MYGUI/MyGUI_FontData.h>
+#include <MYGUI/MyGUI_FontManager.h>
 #endif // USE_MYGUI
 
 #include <OgreTextAreaOverlayElement.h>
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir rigsofrods-source-0.4.0.4.orig/source/main/utils/Utils.h rigsofrods-source-0.4.0.4/source/main/utils/Utils.h
--- rigsofrods-source-0.4.0.4.orig/source/main/utils/Utils.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/utils/Utils.h	2012-09-23 18:11:02.199025777 +0400
@@ -27,7 +27,7 @@
 #include <OgreDataStream.h>
 
 #ifdef USE_MYGUI
-#include <MyGUI.h>
+#include <MYGUI/MyGUI.h>
 #endif //USE_MYGUI
 
 #include <pthread.h>
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir -x CMakeCache.txtr rigsofrods-source-0.4.0.4.orig/source/main/gui/RTTLayer.cpp rigsofrods-source-0.4.0.4/source/main/gui/RTTLayer.cpp
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/RTTLayer.cpp	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/RTTLayer.cpp	2012-11-19 19:11:08.605788996 +0400
@@ -1,12 +1,12 @@
 // based on the MyGUI UnitTests
 
-#include "MyGUI_LayerItem.h"
+#include "MYGUI/MyGUI_LayerItem.h"
 #include "RTTLayer.h"
-#include "MyGUI_Enumerator.h"
-#include "MyGUI_FactoryManager.h"
-#include "MyGUI_RenderManager.h"
-#include "MyGUI_Gui.h"
-#include "MyGUI_LayerNode.h"
+#include "MYGUI/MyGUI_Enumerator.h"
+#include "MYGUI/MyGUI_FactoryManager.h"
+#include "MYGUI/MyGUI_RenderManager.h"
+#include "MYGUI/MyGUI_Gui.h"
+#include "MYGUI/MyGUI_LayerNode.h"
 
 namespace MyGUI
 {
diff -NurEbBH -x bin -x CMakeCache.txt -x CMakeCache.txt.orig -x CMakeOutput.log -x CMakeCXXCompiler.cmake -x configurations.xml -x Package-Default.bash -x '*.includes' -x build.make -x flags.make -x link.txt -x '*.o' -x nbproject -x Progress -x cmake_install.cmake -x rorconfig.dir -x RoR.dir -x CMakeCache.txtr rigsofrods-source-0.4.0.4.orig/source/main/gui/RTTLayer.h rigsofrods-source-0.4.0.4/source/main/gui/RTTLayer.h
--- rigsofrods-source-0.4.0.4.orig/source/main/gui/RTTLayer.h	2012-09-06 13:04:24.000000000 +0400
+++ rigsofrods-source-0.4.0.4/source/main/gui/RTTLayer.h	2012-11-19 19:09:36.636278420 +0400
@@ -3,9 +3,9 @@
 #ifndef __MYGUI_RTT_LAYER_H__
 #define __MYGUI_RTT_LAYER_H__
 
-#include "MyGUI_Prerequest.h"
-#include "MyGUI_Types.h"
-#include "MyGUI_OverlappedLayer.h"
+#include "MYGUI/MyGUI_Prerequest.h"
+#include "MYGUI/MyGUI_Types.h"
+#include "MYGUI/MyGUI_OverlappedLayer.h"
 
 namespace MyGUI
 {
