diff -NaurEbBH SocketW031026.orig/Makefile.conf SocketW031026/Makefile.conf
--- SocketW031026.orig/Makefile.conf	2012-11-26 23:48:50.000000000 +0400
+++ SocketW031026/Makefile.conf	2012-11-26 23:49:13.980284680 +0400
@@ -10,7 +10,7 @@
 PREFIX_H =$(PREFIX)/include
 
 # Compile flags passed to the compiler
-CFLAGS =-Wall -O3 -ffast-math
+CFLAGS +=-Wall -O3 -ffast-math
 SW_CFLAGS =
 
 # Libs config
diff -NaurEbBH SocketW031026.orig/src/Makefile SocketW031026/src/Makefile
--- SocketW031026.orig/src/Makefile	2012-11-26 23:43:37.000000000 +0400
+++ SocketW031026/src/Makefile	2012-11-26 23:49:40.614735386 +0400
@@ -2,7 +2,7 @@
 
 include ../Makefile.conf
 
-CFLAGS += $(SW_CFLAGS) -fPIC -I./
+CFLAGS += $(SW_CFLAGS) $(CXXFLAGS) -fPIC -I./
 LIBS = $(SW_LIBS)
 
 SW_VER = 031026
